Automattic/wp-calypso
# Gutenberg: refactor API override code

Raised in https://github.com/Automattic/wp-calypso/pull/27002#discussion_r214915412 by @dmsnell 

Make sure that we won't unnecessarily execute `overrideAPIPaths` in `render` multiple times, when in only needs to be called once.

### Testing instructions
1. `localStorage.setItem( 'debug', 'calypso:gutenberg' );`
2. Navigate to `/gutenberg/post/{siteSlug}`.
3. Verify in console that requests are still being issued with correct paths.
4. Reload and test with clean state.
5. `npm run test-client client/gutenberg/editor/api-middleware/test/index.js`
